qutebrowser/browser/downloads.py

```diff
@@ -166,6 +166,7 @@ def get_filename_question(*, suggested_filename, url, parent=None):
     q.title = "Save file to:"
     q.text = "Please enter a location for <b>{}</b>".format(
         html.escape(url.toDisplayString()))
+    q.url = url.toString(QUrl.RemoveUserInfo)
```

Comment:
`url.toString(QUrl.RemovePassword | QUrl.FullyEncoded)` is probably more appropriate here. It's fine to keep the username here, and we should encode special characters for the yanked URL (like `:yank url` does).

Same for all other places where you convert a URL to a string.